src/routes/transactions/entities/transaction.entity.ts

```diff
@@ -0,0 +1,35 @@
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { ExecutionInfo } from './execution-info.entity';
+import { SafeAppInfo } from './safe-app-info.entity';
+import { TransactionInfo } from './transaction-info.entity';
+
+export class Transaction {
+  @ApiProperty()
+  id: string;
+  @ApiProperty()
+  timestamp: number;
+  @ApiProperty()
+  txStatus: string;
+  @ApiProperty()
+  txInfo: TransactionInfo;
```

Comment:
`@nestjs/swagger` requires a little bit more info here to generate the possible schemas for `TransactionInfo`

You can do something like:

```typescript
@ApiProperty({
  oneOf: [
    { $ref: getSchemaPath(CustomTransactionInfo) },
    { $ref: getSchemaPath(SettingsChangeTransaction) },
    { $ref: getSchemaPath(TransferTransactionInfo) },
  ],
})
txInfo: TransactionInfo;
```

The Transaction class however needs to include these models too:

```typescript
@ApiExtraModels(
  CustomTransactionInfo,
  SettingsChangeTransaction,
  TransferTransactionInfo,
)
export class Transaction {
  ...
}
```

This way the correct schemas will be available via the Swagger interface. The same can be done for `ExecutionInfo` in this class.

If you prefer to tackle this in a separate PR it's also ok 👍

Reply:
You're right, good catch! Included here [d985d6c](https://github.com/5afe/safe-client-gateway-nest/pull/197/commits/d985d6cb5043fb4c54881c2cdab35c7007f20306). Thanks for the suggestion!